scripts/build_gh_pages

```diff
@@ -4,71 +4,70 @@ readonly SCRIPTS_DIR="$(dirname "$(readlink -f "$0")")"
+git diff --exit-code || (echo "please commit any pending changes first"; exit 1)
+git diff --cached --exit-code || (echo "please commit any pending changes first"; exit 1)
 
-if [ -z "${CHANGES}" ]; then
-  echo "No doc comment changes found in ${PREV_COMMIT}..${BRANCH}" subdirs "${SRC_SUBDIRS[@]}"
-  exit 0
-fi
-
-git checkout "${BRANCH}"
-readonly BRANCH_SHA1=$(git rev-parse --short HEAD)
+readonly BRANCH_SHA1=$(git rev-parse --short=12 HEAD)
 readonly BRANCH_SUBJECT=$(git log -n 1 --format=format:%s)
 readonly COMMIT_MESSAGE=$(cat <<-END
-Update gh-pages to ${BRANCH} branch at ${BRANCH_SHA1}
+Update gh-pages from ${BRANCH_SHA1}
 
 Auto-generated from commit ${BRANCH_SHA1} ("${BRANCH_SUBJECT}").
 END
 )
 
-tgz_gen() {
+# Create a temporary directory to stage the output of the generation process.
+readonly TARGET_DIR=$(mktemp --directory --tmpdir=/tmp 'project-oak-gh-pages-XXXXXXXXXX')
+
+# Clone the gh-pages branch to the target directory, limiting to one commit.
+git clone git@github.com:project-oak/oak.git --branch=gh-pages --depth=1 "${TARGET_DIR}"
+
+# Remove everything from the target directory. This script is supposed to automatically recreate
+# everything within that directory.
+rm --recursive --force ${TARGET_DIR}/*
+
+readonly SRC_SUBDIRS=(sdk/rust oak/server/rust)
+readonly DOC_SUBDIRS=(sdk server)
+
+doc_gen() {
   local srcdir=$1
   local docdir=$2
-  local tarfile="/tmp/oak-doc-${docdir}-${BRANCH_SHA1}.tgz"
-  # Build Cargo docs and save them off outside the repo
   (
     cd "${srcdir}" || exit
-    rm -rf target/doc
+    # Remove previously generated artifacts, since `cargo doc` only regenerates new or modified
+    # files, but does not remove artifacts generated from now-removed files.
+    rm --recursive --force ./target/doc
     cargo doc --no-deps
     cargo deadlinks
-    cd target/doc || exit
-    tar czf "${tarfile}" ./*
+    cp --recursive ./target/doc "${TARGET_DIR}/${docdir}"
   )
-  echo "${tarfile}"
 }
 
-tarfiles=()
 for ii in "${!SRC_SUBDIRS[@]}"; do
-   tarfiles+=( "$(tgz_gen "${SRC_SUBDIRS[$ii]}" "${DOC_SUBDIRS[$ii]}")" )
+  doc_gen "${SRC_SUBDIRS[$ii]}" "${DOC_SUBDIRS[$ii]}"
 done
 
-# Shift to gh-pages branch and replace contents
-git checkout gh-pages
-
-doc_update() {
-  local docdir=$1
-  local tgzfile=$2
-  rm -rf "${docdir}"
-  mkdir "${docdir}"
-  (
-    cd "${docdir}" || exit
-    tar xzf "${tgzfile}"
-  )
-}
-
-for ii in "${!DOC_SUBDIRS[@]}"; do
-  doc_update "${DOC_SUBDIRS[$ii]}" "${tarfiles[$ii]}"
-done
+cat <<-END > "${TARGET_DIR}/index.html"
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta http-equiv="Refresh" content="0; url=./sdk/oak/index.html" />
+  </head>
+  <body>
+    <p><a href="./sdk/oak/index.html">Oak SDK main page</a></p>
+  </body>
+</html>
+END
 
-# Commit any differences
-git add "${DOC_SUBDIRS[@]}"
-git commit --message="${COMMIT_MESSAGE}"
-git checkout "${BRANCH}"
+(
+  cd "${TARGET_DIR}"
+  # Stage everything for commit, including the `index.html` page.
+  git add .
```

Comment:
Not sure what's happening differently, but this now seems to pick up some zero-length `.lock` [files](https://github.com/project-oak/oak/blob/43b566dec7b4bd2cab23f43620542ad9cb3a8bd1/sdk/.lock) &ndash; any idea why?

Reply:
Ah, that's because the old script did  `tar czf "${tarfile}" ./*` thus missing top-level dot-files.